### PR TITLE
refactor(poetry): Use centralized docker execution

### PR DESCRIPTION
**Pull request by zharinov**
ref #2863

### CODE REVIEW

File: lib/manager/poetry/artifacts.ts [hunk 1]
```diff
@@ -43,28 +43,24 @@ export async function updateArtifacts(
     logger.debug(`Updating ${lockFileName}`);
     const cwd = join(config.localDir, subDirectory);
     const env = getChildProcessEnv();
-    let cmd: string;
+    let cmd = 'poetry';
+    const execOptions: ExecOptions = { cwd, env };
```
Comment (rarkins) [inline lib/manager/poetry/artifacts.ts]: Seems like `env` logic could be moved into `lib/util/exec` if it's always sourced from `getChildProcessEnv()` ?

Comment (ViceIce) [reply lib/manager/poetry/artifacts.ts]: Not all exec call use `getChildProcessEnv` but should i think. So we should add this call to the `util/exec` logic.

**Review by viceice — CHANGES_REQUESTED** [review]
move `getChildProcessEnv` call to `util/exec`

File: lib/manager/poetry/artifacts.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { parse, join } from 'upath';
 import { hrtime } from 'process';
 import { outputFile, readFile } from 'fs-extra';
-import { exec } from '../../util/exec';
+import { exec, ExecOptions } from '../../util/exec';
 import { getChildProcessEnv } from '../../util/env';
```
Comment (rarkins) [inline lib/manager/poetry/artifacts.ts]: This can be removed

**Comment by zharinov** [comment]
Moved to #5144

**Pull request closed without merging**